pytorch/training/docker/1.13/py3/cu117/Dockerfile.gpu

```diff
@@ -162,106 +162,65 @@ RUN ompi_info --parsable --all | grep mpi_built_with_cuda_support:value \
  && chmod +x ~/mambaforge.sh \
  && ~/mambaforge.sh -b -p /opt/conda \
  && rm ~/mambaforge.sh \
- && /opt/conda/bin/conda install -y -c conda-forge \
+ && /opt/conda/bin/mamba install -y -c conda-forge \
     python=$PYTHON_VERSION \
     cython \
     mkl \
     mkl-include \
     parso \
     typing \
+    # force conda to install h5py with openmpi build, not mpich (rubik error)
+    # https://anaconda.org/conda-forge/h5py/files?page=2
+    openmpi \
+    "mpi4py>=3.1.4,<3.2" \
     h5py \
+    openmpi \
     requests \
     pyopenssl \
     libgcc \
     conda-content-trust \
     charset-normalizer \
+    packaging \
+    awscli \
+    boto3 \
+    pybind11 \
+    pyyaml \
     # needed by fastai.distributed
-    accelerate \
- && /opt/conda/bin/conda install -c dglteam -y dgl-cuda11.7 \
- && /opt/conda/bin/conda install -c pytorch -y magma-cuda117 \
- && /opt/conda/bin/conda install -c fastai fastai \
-# fastai conda has a dependency chain of fastai -> spacy -> pathy -> dataclasses
+    accelerate \ 
+    cryptography \
+    cmake \
+    scipy \
+    click \
+    # patch CVE
+    "pillow>=9.4" \
+    psutil \
+    ipython \
+ # pin dgl to 0.9.1 as upstream test has not caught up, currently having accuracy error
+ && /opt/conda/bin/mamba install -c dglteam -y dgl-cuda11.7=0.9.1 \
+ && /opt/conda/bin/mamba install -c pytorch -y magma-cuda117 \
+ && /opt/conda/bin/mamba install -c fastai fastai \
+ # fastai conda has a dependency chain of fastai -> spacy -> pathy -> dataclasses
  # conda pathy installs dataclasses regardless of py version: https://github.com/conda-forge/pathy-feedstock/blob/1e6da43da5c8e11e09c063c79b09236bbacb7780/recipe/meta.yaml#L29
  # pypi solves this issue: https://github.com/justindujardin/pathy/blob/master/requirements.txt#L3
- # dataclasses package is not needed for py>=37
- # install plainly will oversize the image
+ # dataclasses package, which contains CVE, is not needed for py>=37
+ # pypi fastai's dependencies will oversize the image
  && pip uninstall -y dataclasses \
- && /opt/conda/bin/conda clean -ya
+ && /opt/conda/bin/mamba clean -ya
 
 # Conda installs links for libtinfo.so.6 and libtinfo.so.6.2 both
 # Which causes "/opt/conda/lib/libtinfo.so.6: no version information available" warning
 # Removing link for libtinfo.so.6. This change is needed only for ubuntu 20.04-conda, and can be reverted
 # once conda fixes the issue
 RUN rm -rf /opt/conda/lib/libtinfo.so.6
 
-RUN /opt/conda/bin/conda config --set ssl_verify False \
- && pip install --no-cache-dir --upgrade pip --no-cache-dir --trusted-host pypi.org --trusted-host files.pythonhosted.org \
- && ln -s /opt/conda/bin/pip /usr/local/bin/pip3
+RUN pip install --no-cache-dir --upgrade pip --no-cache-dir --trusted-host pypi.org --trusted-host files.pythonhosted.org \
+ && ln -s /opt/conda/bin/pip /usr/local/bin/pip3 \
+ # use pip to install opencv since conda package broke nvidia ssl verification
```

Comment:
Does this solution work https://github.com/NVIDIA/nvidia-docker/issues/1009#issuecomment-1181371694?

Reply:
thanks! Trying out this solution now

Reply:
This fixes the apt-get issue thanks!